src/moodleapi/MoodleApi.ts

```diff
@@ -0,0 +1,45 @@
+interface GetUserResponse {
+  uuid: string
+  jwt: string
+}
+
+export class MoodleApi {
+  private readonly wwwroot: string
+  private readonly sesskey: string
+
+  constructor(wwwroot: string, sesskey: string) {
+    this.wwwroot = wwwroot
+    this.sesskey = sesskey
+  }
+
+  async getUser(): Promise<GetUserResponse> {
+    const methodName = 'local_raise_get_user'
+    const userRequest = {
+      index: 0,
+      methodname: methodName,
+      args: {
+      }
+    }
+    const url = this.getMoodleAjaxUrl(methodName)
+    const response = await fetch(url, { method: 'POST', body: JSON.stringify([userRequest]) })
+    const responseJSON = await response.json()
+
+    if (responseJSON.error !== undefined) {
+      throw new Error(responseJSON.error)
+    }
+
+    return {
+      uuid: responseJSON[0].data.uuid,
+      jwt: responseJSON[0].data.jwt
+    }
+  }
+
+  getMoodleAjaxUrl(endpointMethod: string): string {
```

Comment:
This probably got lost in the shuffle in my first round of comments, but should we make this a `private` method (e.g. `private getMoodleAjaxUrl(endpointMethod: string): string {`)?